### DISCUSSION TITLE
use run shell command in previewers section

### DISCUSSION THREAD

**Discussion by og900aero** [sxyazi/yazi#1484] [1. Q&A]
Is it possible to use shell commands under the previerws section?
Example in yazi.toml:

```
previewers = [
  { mime = "application/zstd", run ='tar tf "$1"' }
]
```

**Comment by AnirudhG07** [comment]
You can always make a small plugin which should work. Check out [rich-preview](https://github.com/AnirudhG07/rich-preview.yazi), maybe it could give some reference.

**Reply by og900aero** [reply]
I’m absolutly stupid in lua language.

**Reply by AnirudhG07** [reply]
maybe I can make a small plugin for you. What are you trying to do exactly?

**Reply by AnirudhG07** [reply]
Oh wait, if you are trying to view the files inside a tar file
You already have [ouch.yazi](https://github.com/ndtoan96/ouch.yazi). Check this out. You can view and create archives with this. Hope it Helps!

**Reply by og900aero** [reply]
> Oh wait, if you are trying to view the files inside a tar file You already have [ouch.yazi](https://github.com/ndtoan96/ouch.yazi). Check this out. You can view and create archives with this. Hope it Helps!

I know, but too much dependencies, and much slower.

**Reply by AnirudhG07** [reply]
Check out [here](https://github.com/AnirudhG07/awesome-yazi?tab=readme-ov-file#previewers) whatever suits your needs. It has some existing plugins for archives previews.

**Reply by og900aero** [reply]
I know. I checked this.

**Comment by sxyazi** [comment]
Now supported via https://github.com/yazi-rs/plugins/tree/main/piper.yazi
